test/python/synthesis/test_linear_synthesis.py

```diff
@@ -139,6 +140,29 @@ def test_synth_lnn_kms(self, num_qubits):
                 dist = abs(q0 - q1)
                 self.assertEqual(dist, 1)
 
+    def test_pmh_section_sizes(self):
```

Comment:
I would be happy if we can add some pseudo-random tests with various values of `section_size` as well as `None`, see e.g  this test:
https://github.com/Qiskit/qiskit/blob/2c3ddf1e6f81c497297b1d3531f20ea62b4cb620/test/python/synthesis/test_linear_synthesis.py#L118

Reply:
Done!